### PR TITLE
codeintel: Update candidate repos for auto-indexing

### PR DESCRIPTION
**Pull request by efritz**
Fixes https://github.com/sourcegraph/sourcegraph/issues/26852.

This PR makes the following high-level changes:

- **The repositories we choose to auto-index now come from indexing policies and does not use search contexts as a back-door way to collect sets of repositories.**
- We now have global limits (configured in site settings) to restrict the blast radius of indexing patterns like `*`
- Update the `PreviewRepositoryFilter` graphql resolver to take multiple patterns and return a paginate-able response
- Better UI!

Disallow global policies:

Screen Shot 2021-11-15 at 3 50 43 PM
Screen Shot 2021-11-15 at 3 50 48 PM

Allowed global policies:

Screen Shot 2021-11-15 at 3 51 31 PM

This PR makes the following lower-level changes:

- Replace `RepoIDsByGlobPattern` with `RepoIDsByGlobPatterns` that takes multiple glob patterns and handles pagination
- Update `UpdateReposMatchingPatterns` to takes a optional maximum number of matching repositories per pattern set (presumably belonging to the same policy)
- Update `SelectRepositoriesForIndexScan` to scan the repos matching configuration policies instead of using search contexts. We'll be able to delete our large corpus search context in prod after this PR becomes effective.
- Adds two new site config flags under `codeIntelAutoIndexing`
- Exposes the site config value `codeIntelAutoIndexingAllowGlobalPolicies` to the UI context

### CODE REVIEW

File: client/web/src/enterprise/codeintel/configuration/useSearchRepositories.tsx [hunk 1]
```diff
@@ -35,9 +41,17 @@ export const usePreviewRepositoryFilter = (pattern: string): SearchRepositoriesR
     )
 
     return {
-        previewResult: {
-            preview: data?.previewRepositoryFilter.map(({ name }) => ({ name })) || [],
-        },
+        previewResult: data
```
Comment (efritz) [inline client/web/src/enterprise/codeintel/configuration/useSearchRepositories.tsx:44]: - [ ] Wrap in `Maybe`

File: client/web/src/enterprise/codeintel/configuration/IndexSettings.tsx [hunk 2]
```diff
@@ -5,51 +5,79 @@ import { Toggle } from '@sourcegraph/branded/src/components/Toggle'
...
-        <div className="mb-4">
-            <label htmlFor="index-commit-max-age">Commit max age</label>
-            <DurationSelect
-                id="index-commit-max-age"
-                value={policy.indexCommitMaxAgeHours ? `${policy.indexCommitMaxAgeHours}` : null}
-                disabled={!policy.indexingEnabled}
-                onChange={value => setPolicy({ ...policy, indexCommitMaxAgeHours: value })}
-            />
-        </div>
+    console.log({ context: window.context })
```
Comment (Numbers88s) [inline client/web/src/enterprise/codeintel/configuration/IndexSettings.tsx:31]: Do we need to log it?

**Review by Numbers88s — APPROVED** [review]
Looks great!

**Review by eseliger — APPROVED** [review]
LGTM don't forget to configure this on dotcom :)

File: client/web/src/enterprise/codeintel/configuration/IndexSettings.tsx (hunk 2 above)
Comment (efritz) [reply client/web/src/enterprise/codeintel/configuration/IndexSettings.tsx:31]:
```suggestion
```

**Pull request merged**